### PR TITLE
internal: update blockchain reactor godoc

### PR DESCRIPTION
**Pull request by alexanderbez**
ref: https://github.com/tendermint/tendermint/issues/4651

### CODE REVIEW

File: internal/blockchain/doc.go [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+	environments.
 
+The v0 blockchain reactor Service has one p2p channel, BlockchainChannel. This
+channel is responsible for handling messages that both request blocks and respond
+to block requests from peers. For every block request from a peer, the reactor
+will execute respondToPeer which will fetch the block from the node's state store
+and respond to the peer. For every block response, the node will add the block
+the it's pool via AddBlock.
+
+Internall, v0 runs a poolRoutine that constantly checks for what blocks it needs
```
Comment (cmwaters) [inline internal/blockchain/doc.go:32]:
```suggestion
Internally, v0 runs a poolRoutine that constantly checks for what blocks it needs
```

File: internal/blockchain/doc.go [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+	verified. However, the current implementation of v2 is not as battle-tested and
+	is known to have various bugs that could make it unreliable in production
+	environments.
 
+The v0 blockchain reactor Service has one p2p channel, BlockchainChannel. This
+channel is responsible for handling messages that both request blocks and respond
+to block requests from peers. For every block request from a peer, the reactor
+will execute respondToPeer which will fetch the block from the node's state store
+and respond to the peer. For every block response, the node will add the block
+the it's pool via AddBlock.
```
Comment (cmwaters) [inline internal/blockchain/doc.go:30]:
```suggestion
to its pool via AddBlock.
```

File: internal/blockchain/doc.go [hunk 3]
```diff
@@ -1,17 +1,36 @@
 /*
-Package blockchain provides two implementations of the fast-sync protocol.
+Package blockchain implements two versions of a reactor Service that are
+responsible for block propagation and gossip between peers. This mechanism is
+more formally known as fast-sync.
 
-- v0 was the very first implementation. it's battle tested, but does not have a
-lot of test coverage.
-- v2 is the newest implementation, with a focus on testability and readability.
+In order for a validator to successfully participate in consensus, it must have
```
Comment (cmwaters) [inline internal/blockchain/doc.go:6]: This is a bit of nit, but it's not just validators participating in consensus. All full nodes can

**Review by cmwaters — APPROVED** [review]
A few minor comments but other than that this looks good.

File: internal/blockchain/doc.go (hunk 3 above)
Comment (alexanderbez) [reply internal/blockchain/doc.go:6]: Yeah, I just didn't feel like making that distinction 😆

**Pull request merged**